### PR TITLE
Fix client table for removed clients.

### PR DESCRIPTION
**Pull request by robertnishihara**
Clients that have been removed have two entries in the client table: one for the insertion, which has all of the information, and one for the deletion, which has no information. This changes the client table to display the insertion information instead of the deletion information.

### CODE REVIEW

**Comment by richardliaw** [comment]
This changes the semantics of `client_table` right?

I think this actually might break Tune; this makes it so that removed nodes still show up in `ray.global_state.client_table` (and it's not clear that `IsInsertion=False` with valid resource specs indicates a deleted entry...)

**Comment by robertnishihara** [comment]
Removed nodes currently (before this PR), show up in the client table. The difference is that the node IP address and things like that also show up for removed nodes (as opposed to being empty strings).

**Comment by robertnishihara** [comment]
We could have `ray.global_state.client_table` return too lists. One of live nodes and one of dead nodes. What do you think about that?

**Comment by robertnishihara** [comment]
Jenkins, retest this please

File: python/ray/experimental/state.py [hunk 1]
```diff
@@ -751,22 +761,13 @@ def cluster_resources(self):
...
-                for key, value in client["Resources"].items():
-                    resources[key] += value
+        live_clients, _ = self.client_table()
+        for client in live_clients:
+            for key, value in client["Resources"].items():
+                resources[key] += value
 
         return dict(resources)
 
-    def _live_client_ids(self):
```
Comment (richardliaw) [inline python/ray/experimental/state.py]: I think you should add a public function for `live_client_info()` that just returns the first item of the `client_table()`. It's a very common pattern, even in this PR, and makes the code much more readable.

File: python/ray/test/test_utils.py [hunk 2]
```diff
@@ -32,8 +32,8 @@ def _wait_for_nodes_to_join(num_nodes, timeout=20):
     """
     start_time = time.time()
     while time.time() - start_time < timeout:
-        client_table = ray.global_state.client_table()
-        num_ready_nodes = len(client_table)
```
Comment (richardliaw) [inline python/ray/test/test_utils.py]: do we also only want live nodes here?

**Comment by richardliaw** [comment]
ping, comments

**Comment by richardliaw** [comment]
ping

**Comment by richardliaw** [comment]
Closing this for now because stale and not on critical path; feel free to reopen.

**Pull request closed without merging**